### PR TITLE
feature/EEMD-174_spinner_emerald

### PR DESCRIPTION
**Pull request by juantobonq**
Create component Spinner,
Create examples of Spinner,
Create UI Test

UI Test:
image

https://user-images.githubusercontent.com/101196521/188001303-6b4611d0-25da-4cf0-aafe-7faa1dd90902.mp4

### CODE REVIEW

File: app/src/androidTest/java/co/condorlabs/emerald/spinner/EmeraldSpinnerPage.kt [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    }
+
+    fun validIfSelectItemIsNotEnabled() = apply {
+        composeRule.onNodeWithText("Select").assertIsNotEnabled()
+    }
+
+    fun validIfErrorMessageIsDisplayed() = apply {
+        composeRule.onNodeWithText("Required").assertIsDisplayed()
+    }
+}
```
Comment (JavierAlfonsoSantiagoSalazar) [inline app/src/androidTest/java/co/condorlabs/emerald/spinner/EmeraldSpinnerPage.kt:60]: blank space

File: app/src/androidTest/java/co/condorlabs/emerald/spinner/EmeraldSpinnerTest.kt [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+                selected = items.first(),
+                label ="Select",
+                enabled = false
+            )
+            .validIfSelectItemIsDisplayed()
+            .validIfSelectItemIsNotEnabled()
+    }
+
+    @Test
+    fun givenItemListWhenTheViewIsCreatedAndTheSpinnerThenShouldErrorMessage() {
```
Comment (JavierAlfonsoSantiagoSalazar) [inline app/src/androidTest/java/co/condorlabs/emerald/spinner/EmeraldSpinnerTest.kt:56]: And the spinner what?

File: app/src/androidTest/java/co/condorlabs/emerald/spinner/EmeraldSpinnerTest.kt [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+            .launchView(
+                items = items,
+                selected = items.first(),
+                error = "Required",
+                label ="Select",
+            )
+            .validIfSelectItemIsDisplayed()
+            .validIfErrorMessageIsDisplayed()
+    }
+}
```
Comment (JavierAlfonsoSantiagoSalazar) [inline app/src/androidTest/java/co/condorlabs/emerald/spinner/EmeraldSpinnerTest.kt:71]: Blank space here

File: app/src/main/java/co/condorlabs/emerald/components/spinner/EmeraldSpinner.kt [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+                    }
+                }
+            }
+        }
+
+        if (error.isNotEmpty()) {
+            EmeraldText(text = error, style = EmeraldTextStyle.Danger, modifier = modifier.padding(horizontal = 10.dp))
+        }
+    }
+}
```
Comment (JavierAlfonsoSantiagoSalazar) [inline app/src/main/java/co/condorlabs/emerald/components/spinner/EmeraldSpinner.kt:112]: Blank space

File: demo/src/main/java/co/condorlabs/demo/screens/samples/SpinnerScreen.kt [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+                .padding(horizontal = 10.dp).fillMaxWidth(),
+            optionsR,
+            selectedOptionText,
+            enabled = false,
+            error = errors.value,
+            label = "Disabled",
+            onSelectedClick = {}
+        )
+    }
+}
```
Comment (JavierAlfonsoSantiagoSalazar) [inline demo/src/main/java/co/condorlabs/demo/screens/samples/SpinnerScreen.kt:66]: Blank space

File: app/src/androidTest/java/co/condorlabs/emerald/spinner/EmeraldSpinnerTest.kt (hunk 3 above)
Comment (juantobonq) [reply app/src/androidTest/java/co/condorlabs/emerald/spinner/EmeraldSpinnerTest.kt:71]: Done

File: app/src/main/java/co/condorlabs/emerald/components/spinner/EmeraldSpinner.kt (hunk 4 above)
Comment (juantobonq) [reply app/src/main/java/co/condorlabs/emerald/components/spinner/EmeraldSpinner.kt:112]: Done

File: app/src/androidTest/java/co/condorlabs/emerald/spinner/EmeraldSpinnerPage.kt (hunk 1 above)
Comment (juantobonq) [reply app/src/androidTest/java/co/condorlabs/emerald/spinner/EmeraldSpinnerPage.kt:60]: Done

File: demo/src/main/java/co/condorlabs/demo/screens/samples/SpinnerScreen.kt (hunk 5 above)
Comment (juantobonq) [reply demo/src/main/java/co/condorlabs/demo/screens/samples/SpinnerScreen.kt:66]: Done

File: app/src/androidTest/java/co/condorlabs/emerald/spinner/EmeraldSpinnerTest.kt (hunk 2 above)
Comment (juantobonq) [reply app/src/androidTest/java/co/condorlabs/emerald/spinner/EmeraldSpinnerTest.kt:56]: Done

**Review by JavierAlfonsoSantiagoSalazar — DISMISSED** [review]

**Review by cljacob — APPROVED** [review]

**Pull request merged**